tools/tcpaccept.py

```diff
@@ -147,6 +155,46 @@
 }
 """
 
+bpf_text_tracepoint = """
+TRACEPOINT_PROBE(sock, inet_sock_set_state)
+{
+    if (args->protocol != IPPROTO_TCP)
+        return 0;
+    u32 pid = bpf_get_current_pid_tgid();
+    // pull in details
+    u16 family = 0, lport = 0;
+    family = args->family;
+    lport = args->sport;
+
+    if (family == AF_INET) {
+        struct ipv4_data_t data4 = {.pid = pid, .ip = 4};
+        data4.ts_us = bpf_ktime_get_ns() / 1000;
+        bpf_probe_read(&data4.saddr, sizeof(data4.saddr), args->saddr);
+        bpf_probe_read(&data4.daddr, sizeof(data4.daddr), args->daddr);
```

Comment:
To access args->saddr/daddr, there is no need for bpf_probe_read which is much more expensive.
The following diff uses __built_memcpy which will be converted to proper loads by llvm compiler.
```
diff --git a/tools/tcpaccept.py b/tools/tcpaccept.py
index 06967ed..51995a9 100755
--- a/tools/tcpaccept.py
+++ b/tools/tcpaccept.py
@@ -169,16 +169,16 @@ TRACEPOINT_PROBE(sock, inet_sock_set_state)
     if (family == AF_INET) {
         struct ipv4_data_t data4 = {.pid = pid, .ip = 4};
         data4.ts_us = bpf_ktime_get_ns() / 1000;
-        bpf_probe_read(&data4.saddr, sizeof(data4.saddr), args->saddr);
-        bpf_probe_read(&data4.daddr, sizeof(data4.daddr), args->daddr);
+        __builtin_memcpy(&data4.saddr, args->saddr, sizeof(data4.saddr));
+        __builtin_memcpy(&data4.daddr, args->daddr, sizeof(data4.daddr));
         data4.lport = lport;
         bpf_get_current_comm(&data4.task, sizeof(data4.task));
         ipv4_events.perf_submit(args, &data4, sizeof(data4));
     } else if (family == AF_INET6) {
         struct ipv6_data_t data6 = {.pid = pid, .ip = 6};
         data6.ts_us = bpf_ktime_get_ns() / 1000;
-       bpf_probe_read(&data6.saddr, sizeof(data6.saddr), args->saddr);
-        bpf_probe_read(&data6.daddr, sizeof(data6.daddr), args->daddr);
+        __builtin_memcpy(&data6.saddr, args->saddr, sizeof(data6.saddr));
+        __builtin_memcpy(&data6.daddr, args->daddr, sizeof(data6.daddr));
         data6.lport = lport;
         bpf_get_current_comm(&data6.task, sizeof(data6.task));
         ipv6_events.perf_submit(args, &data6, sizeof(data6));
```

Reply:
I don't know how to rebase .
I have made a new pull from the master ,and make a new PR #1842 . 
Can this work?

Reply:
If #1842 works,we can close this PR